src/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt

```diff
@@ -51,11 +51,14 @@ set(NATIVECRYPTO_SOURCES
     pal_x509ext.c
 )
 
-if (FEATURE_DISTRO_AGNOSTIC_SSL)
-    if (NOT CMAKE_SYSTEM_NAME STREQUAL Linux)
-        message(FATAL_ERROR "FEATURE_DISTRO_AGNOSTIC_SSL can only be enabled for Linux")
-    endif()
+# Always build portable on macOS because OpenSSL is not a system component
+# and our prebuilts should not assume a specific ABI version for the types
+# that use OpenSSL at runtime.
+if (APPLE)
+    set(FEATURE_DISTRO_AGNOSTIC_SSL True)
```

Comment:
It seems that rather than hardcoding this in the cmake script, we should pass the -portable option to the build-native.sh like we do for Linux and update the build-native.sh to accept that option for OSX too.

Reply:
I thought about that, but my inclination was to change the build from "never portable" to "always portable", and it seems like rather than failing the build when you run `build-native.sh` without `-portable` that I'd just set it to true in here.

If you think that we want to enable the option and update the rpath rewrites to also rpath rewrite 1.1, that's valid.  But, unlike Linux--where I expect the distros to build non-portable--I don't think that we'd really go back to a tightly-linked version for macOS.

Reply:
After thinking about it a bit more, I agree with you.